### PR TITLE
feat: UI additions and bug fixes

### PR DESCRIPTION
**Pull request by nam-t24**
-Fix time check for spotlight events
-Fix firebase messaging error on Safari by adding browser support check
-Add notification prompt at home page on load(displays if browser supports push notification and user hasn't enabled notifications)
-Shows ask permission alert from browser depending on same condition as above
-Move Ask a Question to dashboard
-Add Challenge Cards with prize display to hack center
-Removed console.log that are no longer needed
-Add challenge cards to home page

### CODE REVIEW

**Comment by WillieCubed** [comment]
#### 📝 Changed routes:


| [**…ents/AnnouncementCards**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/AnnouncementCards) | [**…mponents/ChallengeCard**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/ChallengeCard) |
|:-:|:-:|
|<a href="https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/AnnouncementCards">Screenshot of /dashboard/Components/AnnouncementCards</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Dk4ToGjr-ZIve_PbtSMCMXFSrJTqgbTijI-SK-EG5AY4M8ULOCxUBB99FCfRXFQACA04sgsAG0b76J--gqXIKw.png">(view full size)</a> |<a href="https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/ChallengeCard">Screenshot of /dashboard/Components/ChallengeCard</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/mK6d5Bl31e8Ezary3TJD7pjpMneu2Gp00NqSR9ujMGOs1KRZuHkwzRlsFp2ifVngLk6LZ_RnZMNVQB5uzVqBMw.png">(view full size)</a> |



| [**…Components/MentorCard1**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/MentorCard1) | [**…Components/MentorCard3**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/MentorCard3) |
|:-:|:-:|
|<a href="https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/MentorCard1">Screenshot of /dashboard/Components/MentorCard1</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/wnG5MdfleINYSr0xXsy9EK3dXwNuTJmlU8f5YYMoVSTNf3zeklz11tKbO9U9QioMkOnS4pZuco5bNMbhG-Sgag.png">(view full size)</a> |<a href="https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/MentorCard3">Screenshot of /dashboard/Components/MentorCard3</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/Ge0czTauB7iQzQGQVBjJTq29XUyjAxrRjIq-YROXwwZ_9qvFIvIuS6kh-P60oTcgS3V8MvsIMG9Ry9AsaEWZzw.png">(view full size)</a> |



| [**…ard/Components/Sidebar**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/Sidebar) | [**…mponents/SpotlightCard**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/SpotlightCard) |
|:-:|:-:|
|<a href="https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/Sidebar">Screenshot of /dashboard/Components/Sidebar</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/l4z3SpIXHrslWdewbIW_HfQ_Auh_arkZ3E6qoU4RB0wzb4wvkXV51U1QTHJVzv5tuG4PU50JWJEkNeDXSEFYgA.png">(view full size)</a> |<a href="https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/Components/SpotlightCard">Screenshot of /dashboard/Components/SpotlightCard</a><br /><sup><a href="https://images.weserv.nl?url=https://sjc.microlink.io/N3-e4fwndj5nH1ojkl-WE0EA6bLgdiX-zJvbm7_YwxtxkD9gvtzH5ss4lEZ7fDQVIr1FNIII1He6YAQwuFjDCA.png">(view full size)</a> |



And 8 other routes:
- [**/api/challenges**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/api/challenges)
- [**/dashboard/hackerpack**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/hackerpack)
- [**/dashboard**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard)
- [**/dashboard/questions**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/questions)
- [**/dashboard/scan-in**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/scan-in)
- [**/dashboard/submit**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/dashboard/submit)
- [**/**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/)
- [**/profile**](https://hackportal-4fu9ix3m7-acmutd.vercel.app/profile)



Commit 9b1bb0b981bb445a58b1b317f9cf4175b3054d0e (https://hackportal-4fu9ix3m7-acmutd.vercel.app).

File: lib/types.d.ts [hunk 1]
```diff
@@ -157,6 +157,19 @@ type KeynoteSpeaker = {
...
 
+/**
+ *
+ * Represent a challenge in hackcenter
+ *
+ * @param title title of challenge
+ * @param description description of that challenge
+ *
+ */
+type Challenge = {
```
Comment (abdullah50053) [inline lib/types.d.ts:168]: May need to add another param for Prizes and ability to add a newline

File: pages/dashboard/index.tsx [hunk 2]
```diff
@@ -104,6 +109,11 @@ export default function Dashboard(props: {
     var endHour = parseInt(endTimeMilitary.split(':')[0]);
     var endMinute = parseInt(endTimeMilitary.split(':')[1].substring(0, 2));
 
+    console.log('Time');
```
Comment (abdullah50053) [inline pages/dashboard/index.tsx:112]: Remove console.log statements before push

File: pages/index.tsx [hunk 3]
```diff
@@ -35,29 +35,57 @@ export default function Home({ keynoteSpeakers }: { keynoteSpeakers: KeynoteSpea
...
         <title>HackPortal</title>
         <meta name="description" content="A default HackPortal instance" />
         <link rel="icon" href="/favicon.ico" />
       </Head>
+      {/* Notification info pop up */}
+      <div
+        id="popup"
+        className="fixed z-50 md:translate-x-0 translate-x-1/2 w-[22rem] rounded-md px-4 py-2 top-16 md:right-6 right-1/2 bg-red-200 md:text-base text-sm"
+      >
+        Turn on push notifications to stay up to date with events and announcements!
```
Comment (abdullah50053) [inline pages/index.tsx:71]: Add check to see if they have push notifs on already or not

**Review by abdullah50053 — DISMISSED** [review]

**Review by DedsecKnight — APPROVED** [review]

**Review by abdullah50053 — APPROVED** [review]

**Pull request merged**